File: aqvp-admin-service/src/main/java/com/aqvp/platform/admin/service/AuditService.java
package com.aqvp.platform.admin.service;

import com.aqvp.platform.admin.domain.AuditEvent;
import java.util.List;
import java.util.UUID;

public interface AuditService {

    AuditEvent recordEvent(String eventType,
                          String action,
                          UUID actorId,
                          String actorName,
                          String actorRole,
                          String organizationId,
                          String resourceType,
                          UUID resourceId,
                          String resourceName,
                          String previousValues,
                          String newValues,
                          String ipAddress,
                          String deviceInfo);

    List<AuditEvent> searchEvents(String actorName, String resourceType, String fromDate, String toDate);
}

File: frontend/aqvp-web/src/features/audit/services/auditService.ts
import { adminApi } from '@/config/axios';
import { API_ENDPOINTS } from '@/constants/api';

export interface AuditEvent {
  id: string;
  eventType: string;
  action: string;
  actorId?: string;
  actorName?: string;
  actorRole?: string;
  organizationId?: string;
  resourceType?: string;
  resourceId?: string;
  resourceName?: string;
  previousValues?: string;
  newValues?: string;
  ipAddress?: string;
  deviceInfo?: string;
  occurredAt: string;
}

export interface AuditEventFilters {
  actorName?: string;
  resourceType?: string;
  fromDate?: string;
  toDate?: string;
}

export const auditService = {
  searchEvents: async (filters: AuditEventFilters): Promise<AuditEvent[]> => {
    const response = await adminApi.get<AuditEvent[]>(API_ENDPOINTS.ADMIN.AUDIT_EVENTS, {
      params: filters,
    });
    return response.data;
  },
};
